### PR TITLE
adding meeting notes for 3-6-2019

### PR DESCRIPTION
**Pull request by vsoch**
Signed-off-by: Vanessa Sochat <vsochat@stanford.edu>

Here is a test to add meeting notes (via a folder with markdowns) from our calls to the repository. These are from [here](https://hackmd.io/El8Dd2xrTlCaCG59ns5cwg)

### CODE REVIEW

**Comment by vsoch** [comment]
p.s. hackmd.io is the coolest site, ever. It puts Google Docs to shame!

**Comment by zendril** [comment]
+1(00) for the hackmd.io approach..   Markdown format, collaborative editing

**Comment by vbatts** [comment]
oh cool. I think this hackmd is the way to go.
I'm :100: % okay ditching the IRC meetbot and using this hackmd approach.
The only thing about making a PR afterwards is the clunkiness of getting approvals to merge. :thinking: I wonder if there is an automation for that. Like a :shipit: meeting notes thingie.

@vsoch Before I saw this PR, I added https://hackmd.io/El8Dd2xrTlCaCG59ns5cwg?view#Call-notes-from-IRC 
Do you mind appending that here?

**Comment by mikebrow** [comment]
> oh cool. I think this hackmd is the way to go.
> I'm 💯 % okay ditching the IRC meetbot and using this hackmd approach.
> The only thing about making a PR afterwards is the clunkiness of getting approvals to merge. 🤔 I wonder if there is an automation for that. Like a :shipit: meeting notes thingie.
> 
> @vsoch Before I saw this PR, I added https://hackmd.io/El8Dd2xrTlCaCG59ns5cwg?view#Call-notes-from-IRC
> Do you mind appending that here?

hackmd looks really cool for communications.. but yeah then having to make a commit each week, and or to add a comment to the prior meeting.. that seems like a hassle. Would rather just keep it there. 

There are a few export tools.. gist might work for archiving? 

Looks like there can only be one "owner" of the note, that might be a administrative problem for that "one" person, or we could create a functional id? Maybe talk to the OCI leadership about it.

**Comment by vbatts** [comment]
yeah, the "owner" of a gist ought to be under 'opencontainers' then

**Comment by vsoch** [comment]
I think it's important to not lose notes in HackMd, per the issues mentioned above. Gists have the same problem in that we are likely to lose them, whether via permission or link. Given that there are enough people on the call, I think it's reasonable to add the notes right after via pull request. As for the LGTM robot rules, I do think that's a little much for the notes. I'm not keen on how that works, but is it repo specific? Could we make a meeting notes repo that wouldn't have these rules? 

In the long run, I'd give this use case to Github as an example and ask for a solution. You could imagine GitHub providing users with some kind of collaborative markdown that (under some set of rules or timings) can go into a pull request. Or we could reach out to HackMD and ask for them to extend the integration to repos too. Thoughts?

**Comment by vsoch** [comment]
worth asking :) https://twitter.com/vsoch/status/1104085530063577088

**Comment by vsoch** [comment]
Look what I found! https://github.com/atom/xray/tree/master/memo_core 

The main developer works at GitHub! Maybe they have already thought of doing this?

**Comment by jackycute** [comment]
Hi @vsoch, could you try out HackDF (currently in beta), which helps hackers keep documentation fresh, at https://hackmd.io/hackdf.
Authorize HackDF and import documents then you can pull request your collaborative notes. Do let us know how you think!

**Comment by vsoch** [comment]
hey @jackycute this looks cool! It seems like it's the other way around though - the user makes a commit to Github (or pull request) and the webhook triggers the docs online (in your interface) to update. Does it support having a collaborative document (akin to a Google Doc) and then having that be sent into a pull request?

**Comment by jackycute** [comment]
Yes it does!
Follow the steps on page and authorize the repos you like to collaborate.
And click "Import document", it will import all files ends with ".md" or ".markdown" After it done importing, you should be able to see a page listing all the imported notes.
Select any note in the left pane, and click edit on the top right, you will again enter into the realtime editing mode.

The magic is here, the editing page of imported note will have push and pull buttons on the navbar.
螢幕快照 2019-03-10 上午10 38 33
Now you can view the changes from GitHub or push to GitHub as pull request!

Enjoy.

**Comment by vsoch** [comment]
Hey @jackycute could you discuss the permissions model that is required for this to work? For example, the page says that it's going to authorize Webhooks permissions,

image

but in fact it looks like you are deploying a GitHub App (as opposed to OAuth 2 with scope to read/write webhooks)

image

GitHub actions are still [notably in beta](https://developer.github.com/actions/) and so I would be hesitant to give any App permissions to interact with my repos using them (on my behalf).  Based on reading about [apps here](https://developer.github.com/apps/differences-between-apps/#webhooks), why do I need to give the app permission to my personal account if the goal is only to interact with one repository? Do you have a more clean cut / spliced install method where the permissions are truly to just one repository?

In summary, I think the screenshots you've shared look really great, but I'm extremely conservative about giving third party access to any of my (or related organization) code. If you might be able to, myself and the maintainers would greatly appreciate a clean walk through of how the steps you have shown map to permissions for users / organizations, even down to the level of the API calls (many of us have used the GitHub API.

Once I have this understanding, and it's clear that the permissions are scoped well enough to grant, I'd be happy to test out these features.

**Comment by jackycute** [comment]
Thanks for pointing this out @vsoch!
I should have clarify this earlier.

We use GitHub App since GitHub encourage this (https://developer.github.com/apps/migrating-oauth-apps-to-github-apps/)

HackDF consists of two parts:
1. Get notified when there is a doc need to be updated (doc freshness checking)
2. Pull changes from GitHub or Push changes to GitHub in HackMD

Let me list all the permission we're requesting and reasons.

Permissions
---
- Repository administration [**read-only**]: to know the protection branches
- Checks [**read & write**]: we will like to integrate the doc freshness checking process to Checks in the future
- Repository [**read & write**]: to pull files from or push files to authorized repos
- Issues [**read & write**]: we will like to open up issue on doc freshness checking
- Repository metadata [**read-only**]: to list all your and your orgs repos (but GitHub will only return the repos that you've authorized)
- Pull requests [**read & write**]: to open up PR from HackMD or close the PR that HackMD created

User Permissions
---
- Emails [**read-only**]: to send email for doc freshness checking and send the result of importing docs

Subscribe to events
---
- Check suite: to know the Check status that we requested
- Pull request: to know the PR from HackMD have been updated
- Push: to know the repo have new changes

Please don't hesitate to let me know if you have any question 😄

**Comment by vsoch** [comment]
Closing in favor of #17
